pysat/tests/test_utils_io.py

```diff
@@ -680,75 +796,350 @@ def test_add_netcdf4_standards_to_meta(self, caplog, missing):
+        else:
+            pysat.utils.io.inst_to_netcdf(self.testInst, outfile,
+                                          meta_translation=meta_trans)
 
-        # Test the metadata update
-        new_labels = ['Format', 'Var_Type', 'Time_Base', 'Time_Scale',
-                      'MonoTon', 'Depend_0', 'Depend_1', 'Display_Type']
-        assert init_meta != self.testInst.meta
-        assert self.testInst.index.name in self.testInst.meta
-        for var in init_meta.keys():
-            for label in new_labels:
-                assert label not in init_meta[var]
-                assert label in self.testInst.meta[var]
+        # Load file back and test metadata is as expected
+        with netCDF4.Dataset(outfile) as open_f:
+            for var in open_f.variables.keys():
+                test_vars = open_f[var].ncattrs()
+                # Confirm translated labels are in the file
+                # Avoid time variables
+                if 'MonoTon' not in test_vars:
+                    form = open_f[var].getncattr('Format')
+                    for key in meta_trans.keys():
+                        # String data doesn't have fill
+                        if key != 'fill' and (form != 'S1'):
+                            testing.assert_list_contains(meta_trans[key],
+                                                         test_vars)
+
+                # Confirm pre-translation form of label not in file.
+                for mvar in meta_trans.keys():
+                    assert mvar not in test_vars, \
+                        '{:} was written to the netCDF file'.format(repr(mvar))
+
+        # Load file using pysat utilities and test translation of multiple
+        # parameters to a single parameter.
+        inv_trans = {}
+        for key in meta_trans.keys():
+            for var in meta_trans[key]:
+                inv_trans[var] = key
+
+        # Load the file
+        data, meta = pysat.utils.io.load_netcdf(outfile,
+                                                meta_translation=inv_trans,
+                                                pandas_format=self.pformat)
+
+        # Confirm inverse translation worked.
+        attrs = list(meta.attrs())
+        for key in meta_trans.keys():
+            # Confirm inverse-translated label in metadata
+            assert key in attrs
+
+            # Confirm labels used in file that should have been translated
+            # are not present.
+            for var in meta_trans[key]:
+                assert var not in attrs
 
-            assert 'Depend_1' not in init_meta[var]
-            if init_meta[var].children is None:
-                assert self.testInst.meta[var, 'Depend_1'] == ''
-            else:
-                assert self.testInst.meta[
-                    var, 'Depend_1'] in self.testInst.variables
+        return
+
+    @pytest.mark.parametrize('assign_flag', [True, False])
+    def test_meta_processor_to_from_netcdf4(self, assign_flag):
```

Comment:
Confused about the nesting of functions in this test method.  Why are they not at the class level?

Reply:
I'm testing the meta_processor functionality, and that requires providing a function to the meta_processor keyword. The internal functions `to_meta_proc` and `from_meta_proc` only have relevance within the larger test, `test_meta_processor....`. The meta_processor functionality itself doesn't do much, it is just supposed to provide metadata input to the supplied function as well as accept the returned dict. So I test within the `to_...` and `from` to ensure things are a dict, make changes, and then outside that function ensure the changes are applied.

Reply:
I thought that might be the case.  Personally, I'd prefer to have these defined at the class level, as it makes the code easier to read and if, for some reason, another test ends up needing these functions they're easier to find.  It's a style preference, though, so it's your call.

Reply:
Updated.